### ISSUE TITLE
When Cmd-Shift-B (or click) on a var type is too generic

### ISSUE THREAD

**Issue by Zteve** [intellij-rust/intellij-rust#1695]
When revealing the type of a variable (usage not declaration), or a method, generics, though known, are not filled in, so one gets type descriptions using (non-local) `type` clauses or generic parameters, e.g.

HashMap methods like get are reported as returning Option<T> or some such.

I would like IntelliJ-Rust to fill in the generics when they are known in context (as Java does, I believe).

**Comment by matklad** [comment]
The relevant code is `org.rust.ide.presentation.TyKt#tyToString`. It should make use of `typeParameterValues`

**Comment by vlad20012** [comment]
It's the same as Ctrl+shift+P on PC?

**Comment by matklad** [comment]
yep

**Comment by Kobzol** [comment]
```rust
let mut map: HashMap<u32, u32> = HashMap::new();
map.get(&5);
```
The type of the second expression is displayed as `Option<&u32>`, so I think this is resolved.

Although maybe the types could also be filled when suggesting methods in autocomplete? For example
`map.insert(`  suggests `k: K, v: V`, while it could suggest `k: u32, v: u32`.

**Comment by vlad20012** [comment]
> Although maybe the types could also be filled when suggesting methods in autocomplete? For example
map.insert( suggests k: K, v: V, while it could suggest k: u32, v: u32.

There is a separate issue about it (somewhere)


Really today I don't understand what this issue is about. Looks like 2 years ago type inference was very bad or Cmd-Shift-B wasn't work at all... So I'll just close it
